ts/mains/main_node.ts

```diff
@@ -1226,6 +1230,66 @@ ipc.on('media-access', async () => {
   await askForMediaAccess();
 });
 
+// Proxy settings
+let proxyUsername: string | null = null;
+let proxyPassword: string | null = null;
+
+async function applyProxySettings() {
+  try {
+    const enabled = sqlNode.getItemById('proxy-enabled')?.value || false;
+
+    if (!enabled) {
+      // Clear proxy if disabled
+      await session.defaultSession.setProxy({ proxyRules: '' });
+      console.log('Proxy disabled');
+      return;
+    }
+
+    const host = (sqlNode.getItemById('proxy-host')?.value || '') as string;
+    const port = sqlNode.getItemById('proxy-port')?.value || 0;
+    const username = (sqlNode.getItemById('proxy-username')?.value || '') as string;
+    const password = (sqlNode.getItemById('proxy-password')?.value || '') as string;
+
+    if (!host || !port) {
+      console.warn('Proxy enabled but host or port is missing');
+      return;
+    }
```

Comment:
might be worth making those checks more explicit with those (from lodash)

```suggestion
    if (!host || !port || !isString(host) || !isNumber(port)) {
      console.warn('Proxy enabled but host or port is missing');
      return;
    }
```